libs/wai-utilities/src/Network/Wai/Utilities/Server.hs

```diff
@@ -226,13 +203,26 @@ onError g m r k e = liftIO $ do
 
 -- | Log an 'Error' response for debugging purposes.
 logError :: MonadIO m => Logger -> Maybe Request -> Error -> m ()
-logError g r (Error c l m) = liftIO $ Log.debug g logMsg
+logError g mr (Error c l m) = liftIO $ Log.debug g logMsg
   where
     logMsg = field "code" (statusCode c)
            . field "label" l
-           . field "request" (fromMaybe "N/A" (lookupRequestId =<< r))
+           . field "request" (maybe "N/A" reqinfo mr)
            . msg (val "\"" +++ m +++ val "\"")
 
+    -- Using unsafePerformIO here is important because we only want it to run if log level is
```

Comment:
I'd prefer `unsafeInterleaveIO` for things like this I think.

It still runs in regular IO; but defers the actual work until the value is forced (which presumably won't happen if we're at a different logging level); I think it should work here 😄